tox.ini

```diff
@@ -410,10 +410,8 @@ commands_pre =
   opentelemetry-instrumentation: pip install opentelemetry-test-utils@{env:CORE_REPO}\#egg=opentelemetry-test-utils&subdirectory=tests/opentelemetry-test-utils
   opentelemetry-instrumentation: pip install -r {toxinidir}/opentelemetry-instrumentation/test-requirements.txt
 
-  openai: pip install opentelemetry-api@{env:CORE_REPO}\#egg=opentelemetry-api&subdirectory=opentelemetry-api
-  openai: pip install opentelemetry-semantic-conventions@{env:CORE_REPO}\#egg=opentelemetry-semantic-conventions&subdirectory=opentelemetry-semantic-conventions
-  openai: pip install opentelemetry-sdk@{env:CORE_REPO}\#egg=opentelemetry-sdk&subdirectory=opentelemetry-sdk
-  openai: pip install opentelemetry-test-utils@{env:CORE_REPO}\#egg=opentelemetry-test-utils&subdirectory=tests/opentelemetry-test-utils
+  # packages that are released individually should list their dependencies in test-requirements.txt
```

Comment:
Just curious, is this something specific to only packages that are released individually? Theoretically, we use ~= for packages that are considered stable because we don't foresee breaking changes (so we leave pyproject.toml as the min dependency and test with latest dependency version out of confidence of that). We should really be pinning the unstable package dependencies in every package and bumping them to the latest version that we are confident in. Although I do see the benefits of moving requirements to a file, the rest of the builds use tox to install dependencies for them and I'm not sure if we'd want to introduce another mechanism as of yet. I am open to discussing this though! If we were to only apply this min version testing to the openai instrumentation for now, would this unblock you and we can further discuss if we want to expand the principal to other libraries?

Reply:
the special thing about them is that we don't auto-update dependency versions for them and don't release them ~every month.

If we pin version, these package won't work together and won't work with the rest of instrumentations. 

It's also the approach that's already used in this repo - 

https://github.com/open-telemetry/opentelemetry-python-contrib/blob/226258e45757f1ecb046dc496b7e2e363cf2dddf/resource/opentelemetry-resource-detector-azure/pyproject.toml#L29

So I'm not changing anything except testing.

To be on the safe side I guess we need to test both - the lowest and the current - this way we'd be able to catch any breaking changes introduced in `opentelemetry-instrumentation`  AND also correctness of `pyproject.toml`. Let me see if I can quickly do it.

Reply:
Yes we will have a larger discussion on not pinning versions anymore but this will be separate from this PR so we won't block it.